django/thunderstore/api/cyberstorm/views/package_detail.py

```diff
@@ -0,0 +1,152 @@
+from django.db.models import CharField, Count, OuterRef, QuerySet, Subquery, Sum, Value
+from rest_framework import serializers
+from rest_framework.generics import RetrieveAPIView, get_object_or_404
+
+from thunderstore.api.cyberstorm.serializers import (
+    CyberstormPackageCategorySerializer,
+    CyberstormTeamMemberSerializer,
+)
+from thunderstore.api.utils import CyberstormAutoSchemaMixin
+from thunderstore.community.models.package_listing import PackageListing
+from thunderstore.repository.models.package import get_package_dependants
+from thunderstore.repository.models.package_version import PackageVersion
+
+
+class DependencySerializer(serializers.Serializer):
+    """
+    Dependencies of a given PackageVersion, listed in a given Community.
+
+    community_identifier and namespace are not present by default and
+    need to be annotated to the object.
+    """
+
+    community_identifier = serializers.CharField()
+    description = serializers.CharField()
+    icon_url = serializers.CharField(source="icon.url")
+    name = serializers.CharField()
+    namespace = serializers.CharField(source="package.namespace.name")
+    version_number = serializers.CharField()
+
+
+class TeamSerializer(serializers.Serializer):
+    """
+    Minimal information to present the team on package detail view.
+    """
+
+    name = serializers.CharField()
+    members = CyberstormTeamMemberSerializer(many=True)
+
+
+class ResponseSerializer(serializers.Serializer):
+    """
+    Data shown on package detail view.
+
+    Expects an annotated and customized CustomListing object.
+    """
+
+    categories = CyberstormPackageCategorySerializer(many=True)
+    community_identifier = serializers.CharField(source="community.identifier")
+    community_name = serializers.CharField(source="community.name")
+    datetime_created = serializers.DateTimeField(source="package.latest.date_created")
+    dependant_count = serializers.IntegerField(min_value=0)
+    dependencies = DependencySerializer(many=True)
+    description = serializers.CharField(source="package.latest.description")
+    download_count = serializers.IntegerField(min_value=0)
+    download_url = serializers.CharField(source="package.latest.full_download_url")
+    full_version_name = serializers.CharField(source="package.latest")
```

Comment:
Mapping to the actual method would be much preferred over the `__str__` implementation, since this'd be extremely hard to search the other way around ("Where is this \_\_str\_\_ used?") and at least I'd probably miss this & break the API accidentally.

I'd say relying on the `__str__` implementation is generally a bad practice outside of classes where it's obvious what it serializes into (and that it won't change in the future)

Reply:
Addressed in https://github.com/thunderstore-io/Thunderstore/pull/968